src/test/java/uk/gov/justice/digital/hmpps/keyworker/rolemigration/RoleAssignmentsServiceTest.java

```diff
@@ -155,7 +173,9 @@ public void failureDuringAddRolesPreventsRemoveRoles() {
         verify(roleService).removeRole("U2", "MDI", "X");
 
         verify(roleService, atLeastOnce()).findUsersForPrisonHavingRole(any(), any());
-        verifyNoMoreInteractions(roleService);
+
+        verifyEventRaised("MDI", "2", expected);
+        verifyNoMoreInteractions(roleService,telemetryClient);
```

Comment:
normally with unit testing the rule is to only verify one thing.  This method seems to be testing everything - could the telemetry testing be moved to a different method?

Reply:
separated out test to verify call to telemetryClient